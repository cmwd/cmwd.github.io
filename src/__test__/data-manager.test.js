const path = require('path');
const DataManager = require('../data-manager');

const dataDir = path.resolve(__dirname, 'test-data', 'section');

describe('Entitie', () => {
  it('should match snapshoted object', () => {
    const entity = DataManager
      .getEntitieDetails(
        `${dataDir}/section-one/03212017_section.md`, '123_section.md');

    expect(entity).toMatchSnapshot();
  });

  it('should normalize date', () => {
    const entity = DataManager
      .getEntitieDetails(
        `${dataDir}/section-one/03212017_section.md`, '123_section.md');

    expect(entity.start).toBe(1490050800000);
    expect(entity.end).toBe(1492725600000);

  });
});

describe('Traverse directories tree', () => {
  it('should match snapshot', () => {
    const tree = DataManager.readDataDir(dataDir);

    console.log(JSON.stringify(tree, null, 2));
    expect(tree).toMatchSnapshot();
  });
});
