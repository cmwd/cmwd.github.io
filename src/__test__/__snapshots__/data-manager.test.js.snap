exports[`Categories Should create category description object 1`] = `
Object {
  "dateFormat": "MM-YYYY",
  "name": "Section One",
}
`;

exports[`Entitie should match snapshoted object 1`] = `
Object {
  "content": "<p>My role was to convert given design project into a Wordpress custom theme. To meet requirements I had to implement custom plugins both in back-end and client side.</p>
",
  "end": 1492725600000,
  "layout": "position",
  "name": "first-position",
  "start": 1490050800000,
  "tags": Array [
    "Wordpress",
    "Timber",
    "SASS",
    "ES6",
  ],
  "url": "url",
}
`;

exports[`Traverse directories tree should match snapshot 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "content": "",
          "entity": "category-descriptor.md",
          "name": "Section One",
        },
        Object {
          "content": "<p>My role was to convert given design project into a Wordpress custom theme. To meet requirements I had to implement custom plugins both in back-end and client side.</p>
",
          "end": 1492725600000,
          "entity": "03212017_section.md",
          "layout": "position",
          "name": "first-position",
          "start": 1490050800000,
          "tags": Array [
            "Wordpress",
            "Timber",
            "SASS",
            "ES6",
          ],
          "url": "url",
        },
        Object {
          "content": "<p>My role was to convert given design project into a Wordpress custom theme. To meet requirements I had to implement custom plugins both in back-end and client side.</p>
",
          "entity": "03212016_section.md",
          "layout": "position",
          "name": "second-position",
          "tags": Array [
            "Wordpress",
            "Timber",
            "SASS",
            "ES6",
          ],
          "url": "url",
        },
      ],
      "entity": "section-one",
    },
  ],
  "entity": "section",
}
`;
